### PR TITLE
Support Cassandra 4.0

### LINKED ISSUE #187: Support Cassandra 4.0
This would make it possible to upgrade a cluster using ecAudit to Cassandra 4.0. After upgrade its possible to switch to the built in audit logging i Cassandra or continue using ecAudit if there are features in ecAudit that are missing in Cassandra that are used.

### PR DESCRIPTION
**Pull request by tommystendahl**
Closes #187

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,13 +51,13 @@
...
 
-        <cassandra.version>3.11.10</cassandra.version>
+        <cassandra.version>4.0.1</cassandra.version>
         <artifact.flavor.version>c3.11</artifact.flavor.version>
 
         <pmd-config-url>pmd-rules.xml</pmd-config-url>
 
         <commons-cli.version>1.4</commons-cli.version>
-        <chronicle-queue.version>4.16.5</chronicle-queue.version>
+        <chronicle-queue.version>5.20.123</chronicle-queue.version>
```
Comment (eperott) [inline pom.xml:60]: I'm thinking it should be possible to remove this dependency declaration and just use whatever version Cassandra 4.x provides.

**Comment by eperott** [comment]
On a general note - I believe there should be a fair bit of documentation to update as well in this patch.

**Comment by eperott** [comment]
Did you explore if it would make sense to make ecAudit more of a plug-in for the audit-feature of Cassandra 4.0? Not entirely sure if it makes sense or if it would allow us to still provide all the ecAudit-specific features. But perhaps worth considering?

File: pom.xml (hunk 1 above)
Comment (tommystendahl) [reply pom.xml:60]: Yes, that makes sens, I will try that.

**Comment by tommystendahl** [comment]
> On a general note - I believe there should be a fair bit of documentation to update as well in this patch.

File: integration-test-allow-all-authorizer/src/test/resources/cassandra.yaml [hunk 2]
```diff
@@ -122,6 +122,23 @@ authorizer: com.ericsson.bss.cassandra.ecaudit.auth.AuditAuthorizer
 #   increase system_auth keyspace replication factor if you use this role manager.
 role_manager: com.ericsson.bss.cassandra.ecaudit.auth.AuditRoleManager
 
+# Network authorization backend, implementing INetworkAuthorizer; used to restrict user
```
Comment (eperott) [inline integration-test-allow-all-authorizer/src/test/resources/cassandra.yaml:125]: Perhaps you already discovered this, but there is a helpful script in bin/import_cassandra_yaml_to_it.sh which can help you to to update config files for integration tests. Just thought I'd mention it.

Comment (eperott) [reply integration-test-allow-all-authorizer/src/test/resources/cassandra.yaml:125]: And perhaps the script need some updates to get things right?

Comment (tommystendahl) [reply integration-test-allow-all-authorizer/src/test/resources/cassandra.yaml:125]: No, I was not aware of that script, I will look into that, thanks.

**Pull request closed without merging**